mne/encoding/model.py

```diff
@@ -0,0 +1,325 @@
+        self.y_ = y
+        return self
+
+    def predict(self, X, preproc_x=None):
+        """Generate predictions using fit coefficients.
+
+        This uses the predict method of the final estimator in the
+        `est` attribute.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input feature array.
+        """
+        # Preprocess X with the pipeline
+        if preproc_x is not None:
+            X = preproc_x.fit_transform(X)
+        return self.est.predict(X)
+
+
+def _check_estimator(est):
+    from sklearn.linear_model import Ridge
+    from sklearn.pipeline import Pipeline
+    if est is None:
+        est = Ridge()
+    elif isinstance(est, string_types):
+        if est not in estimator_dict.keys():
+            raise ValueError("No such solver: {0}".format(est))
+        est = estimator_dict[est]()
+
+    if not is_regressor(est):
+        warn("Custom estimators should have a `fit` and `predict` method,"
+             " and should produce continuous output")
+
+    # Make sure we have a pipeline
+    if not isinstance(est, Pipeline):
+        est = Pipeline([('est', est)])
+    return est
+
+
+def _check_preproc(est):
+    from sklearn.pipeline import Pipeline
+    if est is None:
+        return est
+    if not isinstance(est, Pipeline):
+        raise ValueError('preproc must be a sklearn Pipeline or None')
+    if not hasattr(est, 'transform'):
+        raise ValueError('preproc must have a transform method')
+    return est
+
+
+def pull_feature_names(est, name_attr='names_', base_names=None):
+    """Iterate through steps of a Pipeline and extract feature names.
+
+    Incomplete, but something like this could be used to keep
+    track of feature expansions that took place in a Pipeline (e.g., lags)
+    """
+    if not isinstance(est, Pipeline):
+        raise ValueError('est must be an instance of Pipeline')
+    all_names = []
+    for nm, step in est.steps:
+        if hasattr(step, name_attr):
+            all_names.append(getattr(step, name_attr))
+    return all_names
+
+
+# Custom string-supported estimators
+class CholeskySolver(BaseEstimator):
+    """Solves a linear system with Cholesky decomposition.
+
+    This will work with sparse inputs to save memory.
+    """
+    def __init__(self):
+        self._estimator_type = 'regressor'  # For sklearn compatibility
+
+    def fit(self, X, y=None):
+        """Fit the linear model.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input array to fit.
+        y : array, shape (n_samples, n_outputs)
+            The output array to fit.
+        """
+        a = (X.T * X).toarray()  # dot product of sparse matrices
+
+        self.coef_ = linalg.solve(a, X.T * y, sym_pos=True,
+                                  overwrite_a=True, overwrite_b=True).T
+        return self
+
+    def predict(self, X):
+        return np.dot(self.coef_, X)
```

Comment:
I continued w/ the current Cholesky code just because I didn't want to change a bunch of work that had already been put into the code that was there. Happy to incorporate these changes if other people are on board.

I agree that the sklearn should be fine (and actually I think that it handles things like sparse matrices and such). That said, I don't think the `LinearRegression` class has multiple solvers in the same way that `Ridge` does. If the inputs are sparse then it basically calls `sp.sparse.linalg.lsqr`. So perhaps better to include it as `Ridge` with alpha == 0?